common/src/main/scala/org/apache/celeborn/common/metrics/sink/JsonServlet.scala

```diff
@@ -0,0 +1,316 @@
+    updateInnerMetrics(
+      absSource,
+      new MetricData(
+        s"${nh.name}_999thPercentile",
+        absSource.reportNanosAsMills(snapshot.get999thPercentile),
+        timestamp,
+        labelNames,
+        labelValues))
+  }
+
+  def recordTimer(absSource: AbstractSource, nt: NamedTimer): Unit = {
+    val timestamp = System.currentTimeMillis
+    val labelNames = new util.ArrayList[String]
+    val labelValues = new util.ArrayList[String]
+    nt.labels.map { case (k, v) =>
+      labelNames.add(k)
+      labelValues.add(v)
+    }
+    val snapshot = nt.timer.getSnapshot
+    updateInnerMetrics(
+      absSource,
+      new MetricData(s"${nt.name}_Count", nt.timer.getCount, timestamp, labelNames, labelValues))
+    updateInnerMetrics(
+      absSource,
+      new MetricData(
+        s"${nt.name}_Max",
+        absSource.reportNanosAsMills(snapshot.getMax),
+        timestamp,
+        labelNames,
+        labelValues))
+    updateInnerMetrics(
+      absSource,
+      new MetricData(
+        s"${nt.name}_Mean",
+        absSource.reportNanosAsMills(snapshot.getMean),
+        timestamp,
+        labelNames,
+        labelValues))
+    updateInnerMetrics(
+      absSource,
+      new MetricData(
+        s"${nt.name}_Min",
+        absSource.reportNanosAsMills(snapshot.getMin),
+        timestamp,
+        labelNames,
+        labelValues))
+    updateInnerMetrics(
+      absSource,
+      new MetricData(
+        s"${nt.name}_50thPercentile",
+        absSource.reportNanosAsMills(snapshot.getMedian),
+        timestamp,
+        labelNames,
+        labelValues))
+    updateInnerMetrics(
+      absSource,
+      new MetricData(
+        s"${nt.name}_75thPercentile",
+        absSource.reportNanosAsMills(snapshot.get75thPercentile),
+        timestamp,
+        labelNames,
+        labelValues))
+    updateInnerMetrics(
+      absSource,
+      new MetricData(
+        s"${nt.name}_95thPercentile",
+        absSource.reportNanosAsMills(snapshot.get95thPercentile),
+        timestamp,
+        labelNames,
+        labelValues))
+    updateInnerMetrics(
+      absSource,
+      new MetricData(
+        s"${nt.name}_98thPercentile",
+        absSource.reportNanosAsMills(snapshot.get98thPercentile),
+        timestamp,
+        labelNames,
+        labelValues))
+    updateInnerMetrics(
+      absSource,
+      new MetricData(
+        s"${nt.name}_99thPercentile",
+        absSource.reportNanosAsMills(snapshot.get99thPercentile),
+        timestamp,
+        labelNames,
+        labelValues))
+    updateInnerMetrics(
+      absSource,
+      new MetricData(
+        s"${nt.name}_999thPercentile",
+        absSource.reportNanosAsMills(snapshot.get999thPercentile),
+        timestamp,
+        labelNames,
+        labelValues))
+  }
+
+  private def updateInnerMetrics(absSource: AbstractSource, metricData: MetricData): Unit = {
+    innerMetrics.synchronized {
+      if (innerMetrics.size() >= absSource.metricsCapacity) {
```

Comment:
if condition meet, can we directly return? as remove and add new metricData seems useless

Reply:
> if condition meet, can we directly return? as remove and add new metricData seems useless

Thank you for your suggestion. I have resubmitted, please help me review it again.

Reply:
> if condition meet, can we directly return? as remove and add new metricData seems useless

@RexXiong   Thank you for your suggestion. I have resubmitted, please help me review it again.